server/src/main/java/org/elasticsearch/action/bulk/FailureStoreMetrics.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.bulk;
+
+import org.elasticsearch.telemetry.metric.LongCounter;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+
+import java.util.Map;
+
+/**
+ * A class containing APM metrics for failure stores. See the JavaDoc on the individual methods for an explanation on what they're tracking.
+ * General notes:
+ * <ul>
+ *     <li>When a document is rerouted in a pipeline, the destination data stream is used for the metric attribute(s).</li>
+ * </ul>
+ */
+public class FailureStoreMetrics {
+
+    public static final FailureStoreMetrics NOOP = new FailureStoreMetrics(MeterRegistry.NOOP);
+
+    public static final String METRIC_TOTAL = "es.data_stream.ingest.documents.total";
+    public static final String METRIC_FAILURE_STORE = "es.data_stream.ingest.documents.failure_store.total";
+    public static final String METRIC_REJECTED = "es.data_stream.ingest.documents.rejected.total";
+
+    private final LongCounter totalCounter;
+    private final LongCounter failureStoreCounter;
+    private final LongCounter rejectedCounter;
+
+    public FailureStoreMetrics(MeterRegistry meterRegistry) {
+        totalCounter = meterRegistry.registerLongCounter(METRIC_TOTAL, "total number of documents that were sent to a data stream", "unit");
+        failureStoreCounter = meterRegistry.registerLongCounter(
+            METRIC_FAILURE_STORE,
+            "number of documents that got redirected to the failure store",
+            "unit"
+        );
+        rejectedCounter = meterRegistry.registerLongCounter(METRIC_REJECTED, "number of documents that were rejected", "unit");
+    }
+
+    /**
+     * This counter tracks the number of documents that we <i>tried</i> to index into a data stream. This includes dropped documents.
```

Comment:
I would suggest to rephrase the last sentence to something more specific. If I am unfamiliar with the drop processor I would not have understood what you meant.
`This includes dropped documents.` -> `This includes documents that were "dropped" by a pipeline.`.